include/zephyr/drivers/clock_control.h

```diff
@@ -43,7 +43,6 @@ enum clock_control_status {
 	CLOCK_CONTROL_STATUS_STARTING,
 	CLOCK_CONTROL_STATUS_OFF,
 	CLOCK_CONTROL_STATUS_ON,
-	CLOCK_CONTROL_STATUS_UNAVAILABLE,
```

Comment:
I have removed this as the only in-tree usage was to indicate a non-ready clock control device. We can leave it in if people suspect out-of-tree users, but I believe `CLOCK_CONTROL_STATUS_UNKNOWN` would be good enough for such usage.